### PR TITLE
feat: upgrade go to 1.23

### PR DESCRIPTION
**Pull request by StarpTech**
## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/actions/build-push-image/action.yaml [hunk 1]
```diff
@@ -25,6 +25,9 @@ inputs:
   push:
     description: "Whether to push the image to the GitHub container registry"
     default: "true"
+  load_Image:
+    description: "Whether to make the image available to the build. Doesn't work multi-platform builds"
```
Comment (Noroth) [inline .github/actions/build-push-image/action.yaml:29]: nit: 
```suggestion
    description: "Whether to make the image locally available to the build. Doesn't work with multi-platform builds"
```

File: .github/workflows/router-ci.yaml [hunk 2]
```diff
@@ -184,12 +184,16 @@ jobs:
           image_name: router
           image_description: "Cosmo Router"
           image_platforms: 'linux/amd64'
+          load_Image: 'true'
           push: 'false'
 
+      - run: podman images
+
```
Comment (Noroth) [inline .github/workflows/router-ci.yaml:191]: Looks like a debug instruction. Can this be removed?

File: connect-go/go.sum [hunk 3]
```diff
@@ -0,0 +1,10 @@
+connectrpc.com/connect v1.17.0 h1:W0ZqMhtVzn9Zhn2yATuUokDLO5N+gIuBWMOnsQrfmZk=
+connectrpc.com/connect v1.17.0/go.mod h1:0292hj1rnx8oFrStN7cB4jjVBeqs+Yx5yDIC2prWDO8=
+github.com/google/go-cmp v0.5.9 h1:O2Tfq5qg4qc4AmwVlvv0oLiVAGB7enBSJ2x2DqQFi38=
+github.com/google/go-cmp v0.5.9/go.mod h1:17dUlkBOakJ0+DkrSSNjCkIjxS6bF9zb3elmeNGIjoY=
+golang.org/x/net v0.23.0 h1:7EYJ93RZ9vYSZAIb2x3lnuvqO5zneoD6IvWjuhfxjTs=
+golang.org/x/net v0.23.0/go.mod h1:JKghWKKOSdJwpW2GEx0Ja7fmaKnMsbu+MWVZTokSYmg=
```
Comment (Noroth) [inline connect-go/go.sum:6]: This might pop up in our security tab at some point.

**Review by Noroth — APPROVED** [review]
Only some nits. Your call

File: .github/workflows/router-ci.yaml (hunk 2 above)
Comment (StarpTech) [reply .github/workflows/router-ci.yaml:191]: Good catch

File: connect-go/go.sum (hunk 3 above)
Comment (StarpTech) [reply connect-go/go.sum:6]: That should have catched already.

**Pull request merged**